parsons/databases/bigquery/bigquery.py

```diff
@@ -0,0 +1,968 @@
+        try:
+            yield conn
+        finally:
+            conn.close()
+
+    @contextmanager
+    def cursor(self, connection):
+        cur = connection.cursor()
+        try:
+            yield cur
+        finally:
+            cur.close()
+
+    def query(
+        self,
+        sql: str,
+        parameters: Optional[Union[list, dict]] = None,
+        return_values: bool = True,
+    ) -> Optional[Table]:
+        """
+        Run a BigQuery query and return the results as a Parsons table.
+
+        To include python variables in your query, it is recommended to pass them as parameters,
+        following the BigQuery style where parameters are prefixed with `@`s.
+        Using the ``parameters`` argument ensures that values are escaped properly, and avoids SQL
+        injection attacks.
+
+        **Parameter Examples**
+
+        .. code-block:: python
+
+           name = "Beatrice O'Brady"
+           sql = 'SELECT * FROM my_table WHERE name = %s'
+           rs.query(sql, parameters=[name])
+
+        .. code-block:: python
+
+           name = "Beatrice O'Brady"
+           sql = "SELECT * FROM my_table WHERE name = %(name)s"
+           rs.query(sql, parameters={'name': name})
+
+        `Args:`
+            sql: str
+                A valid BigTable statement
+            parameters: dict
+                A dictionary of query parameters for BigQuery.
+
+        `Returns:`
+            Parsons Table
+                See :ref:`parsons-table` for output options.
+        """
+
+        with self.connection() as connection:
+            return self.query_with_connection(
+                sql, connection, parameters=parameters, return_values=return_values
+            )
+
+    def query_with_connection(
+        self, sql, connection, parameters=None, commit=True, return_values: bool = True
+    ):
+        """
+        Execute a query against the BigQuery database, with an existing connection.
+        Useful for batching queries together. Will return ``None`` if the query
+        returns zero rows.
+
+        `Args:`
+            sql: str
+                A valid SQL statement
+            connection: obj
+                A connection object obtained from ``redshift.connection()``
+            parameters: list
+                A list of python variables to be converted into SQL values in your query
+            commit: boolean
+                Must be true. BigQuery
+
+        `Returns:`
+            Parsons Table
+                See :ref:`parsons-table` for output options.
+        """
+
+        if not commit:
+            raise ValueError(
+                """
+                BigQuery implementation uses an API client which always auto-commits. If you wish to wrap
+                multiple queries in a transaction, use Mulit-Statement transactions within a single query
+                as outlined here: https://cloud.google.com/bigquery/docs/transactions
+            """
+            )
```

Comment:
Kept this param to match with Redshift and then throw with a descriptive error. Could just omit the param altogether